great_expectations/execution_engine/partition_and_sample/data_partitioner.py

```diff
@@ -73,7 +73,7 @@ def __hash__(self: PartitionerMethod):
         return hash(self.value)
 
 
-class DataPartitioner(abc.ABC):
+class DataPartitioner(abc.ABC):  # noqa: B024
```

Comment:
https://docs.astral.sh/ruff/rules/abstract-base-class-without-abstract-method/
# abstract-base-class-without-abstract-method (B024)

Derived from the **flake8-bugbear** linter.

## What it does
Checks for abstract classes without abstract methods.

## Why is this bad?
Abstract base classes are used to define interfaces. If an abstract base
class has no abstract methods, you may have forgotten to add an abstract
method to the class or omitted an `@abstractmethod` decorator.

If the class is _not_ meant to be used as an interface, consider removing
the `ABC` base class from the class definition.

## Example
```python
from abc import ABC


class Foo(ABC):
    def method(self):
        bar()
```

Use instead:
```python
from abc import ABC, abstractmethod


class Foo(ABC):
    @abstractmethod
    def method(self):
        bar()
```

## References
- [Python documentation: `abc`](https://docs.python.org/3/library/abc.html)

Reply:
I'm fine with it as is, even though ABCs without abstractmethods can be instantiated directly. We could also do something like suggested [here](https://stackoverflow.com/questions/7989042/preventing-a-class-from-direct-instantiation-in-python), but I think it should be apparent to readers that they shouldn't directly instantiate this, which is the most important thin IMO